File: 701615-tie-breaker-GFG/701615-tie-breaker.cpp
#include <unordered_map>
#include <string>
using namespace std;

class Solution {
public:
    string tieBreak(vector<string>& arr) {
        unordered_map<string, int> mp;

        for (string &name : arr) {
            mp[name]++;
        }

        string ans = "";
        int maxVotes = 0;

        for (auto &it : mp) {
            if (it.second > maxVotes) {
                maxVotes = it.second;
                ans = it.first;
            } 
            else if (it.second == maxVotes && it.first < ans) {
                ans = it.first;
            }
        }

        return ans;
    }
};

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 701615-tie-breaker-GFG/701615-tie-breaker.java
import java.util.*;

class Solution {
    public String tieBreak(String arr[]) {
        HashMap<String, Integer> map = new HashMap<>();

        for (String name : arr) {
            map.put(name, map.getOrDefault(name, 0) + 1);
        }

        String ans = "";
        int maxVotes = 0;

        for (String name : map.keySet()) {
            int votes = map.get(name);

            if (votes > maxVotes) {
                maxVotes = votes;
                ans = name;
            } else if (votes == maxVotes && name.compareTo(ans) < 0) {
                ans = name;
            }
        }

        return ans;
    }
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna